src/iden3comm/handlers/auth.ts

```diff
@@ -517,4 +520,42 @@ export class AuthHandler
     }
     return mediaType;
   }
+
+  private getDefaultProvingMethodAlg(
+    preferredAuthProvingMethod?: ProvingMethodAlg,
+    accept?: string[]
+  ): ProvingMethodAlg {
+    if (
+      preferredAuthProvingMethod &&
```

Comment:
we still should check that preferredAuthProvingMethod persists in  `accept` arr.